### PR TITLE
Added session_log events to text_log

### PR DESCRIPTION
**Pull request by Demilivor**
Request to add logs come here:
https://github.com/ClickHouse/ClickHouse/issues/51760#issuecomment-1652954316

Added session logout to the text log.
Example:
`TCP-Session: eb9b9e9c-a978-4cfd-9f84-cf25eb286601 Logout`

Added authentication failure log to the text_log in the case of Postgres and MySQL sessions:
`MYSQL-Session: 9a243dc2-6503-4b8a-8ded-fc1821c98503 Authentication failed with an error: default: Authentication failed: password is incorrect, or there is no user with such name.`
from `Session::getAuthenticationTypeOrLogInFailure`

Note this error is also duplicated in such logs:

`
MySQLHandler: DB::Exception: default: Authentication failed: password is incorrect, or there is no user with such name.
`

But those logs come from `MySQLHandler` logger not from `MYSQL-Session` and those logs do not contain `auth_id` (`9a243dc2-6503-4b8a-8ded-fc1821c98503`)

TCP, HTTP, and GRPC session login failures can be associated with such a text log entry:
`TCP-Session: 0e3de6fd-9b56-4718-a154-ce3e8963bb4e Authentication failed with an error: invalid: Authentication failed: password is incorrect, or there is no user with such name`

Session login success can be associated with such log:
`TCP-Session: ec527ab3-7427-4233-99a3-f8d083c1fa47 Authenticating user 'default' from [::1]:56210`

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
This is an automated comment for commit 0fc9d62ae47188e3a16c1a517ced127adade5064 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/53142/0fc9d62ae47188e3a16c1a517ced127adade5064/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by tavplubix — APPROVED** [review]

**Comment by Demilivor** [comment]
I updated test `00002_log_and_exception_messages_formatting` (it is failing) expectations:

```
messages shorter than 16	4
exceptions shorter than 30	3
```

Because the new Logout message is very short:
`2023-08-08 13:56:36 17262	0.02	724.87 KiB	0.007	4	169	['Debug']	0.995	{} Logout`

File: tests/queries/0_stateless/00002_log_and_exception_messages_formatting.reference [hunk 1]
```diff
@@ -1,7 +1,7 @@
 runtime messages	0.001
 runtime exceptions	0.05
-messages shorter than 10	1
-messages shorter than 16	3
+messages shorter than 10	2
+messages shorter than 16	4
```
Comment (tavplubix) [inline tests/queries/0_stateless/00002_log_and_exception_messages_formatting.reference:4]: It would be better to fix the message itself

Comment (Demilivor) [reply tests/queries/0_stateless/00002_log_and_exception_messages_formatting.reference]: Done! Added `user_id` to the logout message

**Review by tavplubix — APPROVED** [review]

**Comment by Demilivor** [comment]
This PR depends on another: https://github.com/ClickHouse/ClickHouse/pull/53255. Those changes have been reverted now. Updated changes added to another PR: #53255

**Comment by tavplubix** [comment]
Stateless tests (release, wide parts enabled) - https://github.com/ClickHouse/ClickHouse/pull/53429

**Comment by tavplubix** [comment]
Merged in https://github.com/ClickHouse/ClickHouse/pull/53431

**Pull request closed without merging**